File: ios/ZoomManager/ZoomManager.m
//
//  ZoomManager.m
//  RNZoomDemo
//
//  Created by Carlos Thurber on 1/5/19.
//  Copyright © 2019 Facebook. All rights reserved.
//

#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(RCTZoomManager, NSObject)

RCT_EXTERN_METHOD(
                  initialize:(NSString*)key
                  secret:(NSString*)secret
                  resolver:(RCTPromiseResolveBlock)resolver
                  rejecter:(RCTPromiseRejectBlock)rejecter
                  )

RCT_EXTERN_METHOD(
                  joinMeeting:(NSDictionary *)options
                  resolver:(RCTPromiseResolveBlock)resolver
                  rejecter:(RCTPromiseRejectBlock)rejecter
                  )

@end

File: ios/ZoomManager/ZoomManager.swift
//
//  ZoomManager.swift
//  RNZoomDemo
//
//  Created by Carlos Thurber on 1/5/19.
//  Copyright © 2019 Facebook. All rights reserved.
//

@objc (RCTZoomManager)
class RCTZoomManager: NSObject {

  @objc var methodQueue = DispatchQueue.main

  @objc static func requiresMainQueueSetup() -> Bool {
    return true
  }

  var promiseResolver : RCTPromiseResolveBlock?
  var promiseRejecter : RCTPromiseRejectBlock?

}
